packages/os/warm-pool-wait.service

```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Wait for the instance to be in service
+After=network-online.target configured.target
+Wants=network-online.target configured.target
+# We only want to run once, at first boot. This file is created by shibaken
+# after a successful run.
+ConditionPathExists=!/var/lib/bottlerocket/warm-pool-wait.ran
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/shibaken warm-pool-wait
+RemainAfterExit=true
+
+[Install]
+WantedBy=multi-user.target
```

Comment:
If you take the above suggestion and don't touch `kubelet.service`:

```suggestion
[Unit]
Description=Wait for the instance to be in service
After=configured.target
Wants=configured.target
Before=kubelet.service
# We only want to run once, at first boot. This file is created by shibaken
# after a successful run.
ConditionPathExists=!/var/lib/bottlerocket/warm-pool-wait.ran

[Service]
Type=oneshot
ExecStart=/usr/bin/shibaken warm-pool-wait
RemainAfterExit=true

[Install]
WantedBy=multi-user.target
RequiredBy=kubelet.service
```

(You still don't need to depend on `network-online.target`, it's implied by `configured.target`.)

Reply:
I think we may still need `network-online.target`.  I consistently have this error when we don't depend on `network-online.target` here. I think we mean to make  the `configured.target` depend on `network-online.target`, but it seems the systemd doesn't "wait" for the entire graph of dependencies to start a service. 

```
× warm-pool-wait.service - Wait for the instance to be in service
     Loaded: loaded (/x86_64-bottlerocket-linux-gnu/sys-root/usr/lib/systemd/system/warm-pool-wait.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Tue 2023-01-10 02:43:21 UTC; 4min 33s ago
    Process: 1276 ExecStart=/usr/bin/shibaken warm-pool-wait (code=exited, status=1/FAILURE)
   Main PID: 1276 (code=exited, status=1/FAILURE)

Jan 10 02:43:21 ip-192-168-7-249.us-west-2.compute.internal systemd[1]: Starting Wait for the instance to be in service...
Jan 10 02:43:21 ip-192-168-7-249.us-west-2.compute.internal shibaken[1276]: 02:43:21 [INFO] shibaken started
Jan 10 02:43:21 ip-192-168-7-249.us-west-2.compute.internal shibaken[1276]: autoscaling.should-wait value is true
Jan 10 02:43:21 ip-192-168-7-249.us-west-2.compute.internal shibaken[1276]: Failed to check autoscaling warm pool: IMDS request failed: No 'target-lifecycle-state' found
Jan 10 02:43:21 ip-192-168-7-249.us-west-2.compute.internal systemd[1]: warm-pool-wait.service: Main process exited, code=exited, status=1/FAILURE
Jan 10 02:43:21 ip-192-168-7-249.us-west-2.compute.internal systemd[1]: warm-pool-wait.service: Failed with result 'exit-code'.
Jan 10 02:43:21 ip-192-168-7-249.us-west-2.compute.internal systemd[1]: Failed to start Wait for the instance to be in service.
```

Reply:
Hmm, the dependencies of `early-boot-config.service` alone should be enough to push `preconfigured.target` (and thus `configured.target`) beyond `network-online.target`. I am 99.99% confident here; if I'm wrong then we have bigger issues to fix. 😀 

I don't know why the additional dependency would make a difference here, but my hunch is that the correct fix would be to retry until `target-lifecycle-state` doesn't return a 404, and only then check the returned value.

Otherwise it could be that the dependency is somehow introducing enough of a delay that we "win" a race with a behind-the-scenes update to IMDS data served to the instance.